src/psyclone/psyGen.py

```diff
@@ -3083,6 +3083,129 @@ def scalars(self):
             "Arguments.scalars must be implemented in sub-class")
 
 
+class DataAccess(object):
+    '''A helper class to handle dependencies due to overlapping data
+    accesses between fields
+
+    '''
+
+    def __init__(self, arg):
+        '''store the source argument and its parent call. Also set any
+        internal coverage data.
+
+        :param arg: the argument that we are concerned with
+        :type arg: :py:class:`psyclone.psyGen.Argument`
+
+        '''
+        self._arg = arg
+        self._call = arg.call
+        self.reset_coverage()
+
+    def overlaps(self, arg):
+        '''Determine whether the accesses to the provided field overlap
+        with the accesses of the source argument
+
+        We do not currently deal with accesses to a subset of a field
+        (unless it is a vector). This distinction will need to be
+        added once loop splitting is supported and would be useful for
+        boundary condition kernels
+
+        :param arg: the argument to compare with our internal argument
+        :type arg: :py:class:`psyclone.psyGen.Argument`
+        :return Bool: True is there are overlapping accesses between \
```

Comment:
s/Bool/bool/ and again, please explain what 'overlapping' means.

Reply:
Tried to.